### PR TITLE
Ingestion status callback update

### PR DESCRIPTION
**Pull request by yassinsws**
This PR modifies the ingestion Stucture to be able to update artemis successfully, when ingestion is finished.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new webhook endpoint for deleting lectures, enhancing the deletion process.
	- Added a new data transfer object for managing lecture deletions.
	- Implemented a new status callback class specifically for lecture deletion tracking.
	- Added an API endpoint to retrieve the ingestion state of lecture units.

- **Improvements**
	- Simplified method signatures for deleting old lectures, improving flexibility and modularity.
	- Enhanced error handling for lecture deletion operations, providing clearer status updates.

- **Bug Fixes**
	- Removed unnecessary fields from existing classes to ensure cleaner data handling, improving overall robustness.

### CODE REVIEW

File: app/web/routers/ingestion_status.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+router = APIRouter(prefix="/api/v1", tags=["ingestion_status"])
+
+
+@router.get(
+    "/courses/{course_id}/lectures/{lecture_id}/lectureUnits/{lecture_unit_id}/ingestion-state",
+    dependencies=[Depends(TokenValidator())],
+)
+def get_lecture_unit_ingestion_state(
+    course_id: int, lecture_id: int, lecture_unit_id: int, base_url: str
+):
```
[the remark this answers is not in the document]
Comment (kaancayli) [reply app/web/routers/ingestion_status.py:19]: @yassinsws You may want to address this.

File: app/web/routers/ingestion_status.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        limit=1,
+        return_properties=[LectureSchema.LECTURE_UNIT_NAME.value],
+    )
+
+    if len(result.objects) > 0:
+        return Response(
+            status_code=status.HTTP_200_OK,
+            content='"DONE"',
+            media_type="application/json",
+        )
```
[the remark this answers is not in the document]
Comment (kaancayli) [reply app/web/routers/ingestion_status.py:40]: @yassinsws Here, there seems to be mismatch with the actual format of the content returned and the specified media type. Content seems like a string literal but media type is specified as `application/json`.  We should either use `text/plain` as media type or return a proper JSON respone.

File: app/web/routers/ingestion_status.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            )
+        ),
+        limit=1,
+        return_properties=[LectureSchema.LECTURE_UNIT_NAME.value],
+    )
+
+    if len(result.objects) > 0:
+        return Response(
+            status_code=status.HTTP_200_OK,
+            content='"DONE"',
```
Comment (kaancayli) [inline app/web/routers/ingestion_status.py:38]: Ingestion states can be extracted to an Enum, instead of hardcoding.

**Review by kaancayli — CHANGES_REQUESTED** [review]
Code looks good. Left some comments.

**Review by MichaelOwenDyer — APPROVED** [review]
Code looks good

**Review by kaancayli — APPROVED** [review]
reapprove

**Pull request merged**